include/dcsam/DCSAM.h

```diff
@@ -70,14 +70,24 @@ class DCSAM {
    * factors to add.
    * @param dcfg - a DCFactorGraph containing any joint discrete-continuous
    * factors to add.
-   * @param initialGuess - an initial guess for any new continuous keys that.
-   * appear in the updated factors (or if one wants to force override previously
-   * obtained continuous values).
+   * @param initialGuessContinuous - an initial guess for any new continuous 
+   * keys that appear in the updated factors (or if one wants to force override
+   * previously obtained continuous values).
+   * @param initialGuessDiscrete - an initial guess for any new discrete keys
+   * that appear in the updated factors (or if one wants to force override
+   * previously obtained discrete values).
+   * @param removeFactorIndices - indices of continuous factors to remove
+   * @param removeDiscreteFactorIndices - indices of discrete factors to remove
+
    */
   void update(const gtsam::NonlinearFactorGraph &graph,
               const gtsam::DiscreteFactorGraph &dfg, const DCFactorGraph &dcfg,
               const gtsam::Values &initialGuessContinuous = gtsam::Values(),
-              const DiscreteValues &initialGuessDiscrete = DiscreteValues());
+              const DiscreteValues &initialGuessDiscrete = DiscreteValues(),
+              const gtsam::FactorIndices &removeFactorIndices =
+                                                     gtsam::FactorIndices(),
+              const std::vector<size_t> &removeDiscreteFactorIndices = 
+                                                            std::vector<size_t>());
```

Comment:
Quick question on this: why `std::vector<size_t>` and not `gtsam::FactorIndices` for consistency?

Reply:
The DiscreteFactorGraph removal function takes in a `size_t` whereas the FactorIndices is used to remove factors in isam, so there is a decision about whether we should be consistent with gtsam or self-consistent within DCSAM... Happy to change to `gtsam::FactorIndices` if you think that's best, I initially implemented everything trying to keep as close to the GTSAM interface as possible.